src/vs/editor/contrib/inlayHints/inlayHintsController.ts

```diff
@@ -145,6 +147,9 @@ export class InlayHintsController implements IEditorContribution {
 		const fontFamilyVar = '--inlayHintsFontFamily';
 		this._editor.getContainerDomNode().style.setProperty(fontFamilyVar, fontFamily);
 
+		const key = this._contextKeyService.getContextKeyValue('config.editor.useInjectedText');
+		const shouldUseInjectedText = key === undefined ? true : !!key;
```

Comment:
There are no real customers yet, you won't break anyone so I would just go for it and not have a setting. In case you insist on a setting, use the `IConfigurationService` to get its value, not the context key service

Reply:
I already use this setting for inline completions. It is a emergency exit if there is an horrible not-yet discovered bug in injected text.
I intend to remove it once I can sufficiently rule out there is no such bug (probably once it has been in stable for one milestone), so I didn't bother importing the IConfigurationService. I actually got the idea from you to use the context key service 😉

Reply:
> It is an emergency exit if there is an horrible not-yet discovered bug in injected text.

If you turn it on by default these bugs will be discovered before they turn into emergencies. That's what we have insiders for 💪 

> IConfigurationService. I actually got the idea from you to use the context key service 😉

There must have been a misunderstanding.. Yeah, you can read config values via context keys but it shouldn't be used instead of the config service. Esp not if you need to import a new service anyways.